### DISCUSSION TITLE
How best to deploy pyetrade algo at scheduled times on the cloud?

### DISCUSSION THREAD

**Discussion by nathanramoscfa** [jessecooper/pyetrade#74] [Q&A]
I have been working with Windows Task Manager locally and Google Cloud in the cloud but have been very interested in knowing how others deploy their automated trading programs. Do you do it locally or on the cloud? Which cloud providers do you use? Can you point to any examples of deploying an automated program built with pyetrade?

**Comment by ecourreges** [comment]
Hello, I’ve been using AWS EC2 for 5 years, non automated install (just local dev + rsync to the vm on aws). Costs around 30$/month minimum but is very reliable vs a local hosting. Coming from linux it’s very easy. You can also rent a windows server for a bit more money, which i did at some point to automate clicking on a windows app.
If you want a very shorter latency though I would wait until someone recommends a datacenter closest to New York or to the etrade servers.

In any case i don’t recommend sticking with one trading system, better to have backups.
For example i haven’t used etrade since they got bought back by Morgan Stanley, they cracked down on non-residents due to stricter compliance, which tradestation didn’t do. Also tradestation doesn’t require selenium to get through the authentication, it can be done once and renewed everyday forever.

When i switched to tradestation I added an abstraction layer and unit tests to my auto trading app, to be able to add quickly other brokers with minimal bugs and rewrite.
Unfortunately this doesn’t exist off the ahelf for the trad-fi world, but exists in crypto: https://github.com/ccxt/ccxt

Good luck!

**Reply by nathanramoscfa** [reply]
Google Cloud and AWS seem to have pretty steep learning curves, so for anyone reading this, try Windows Task Scheduler on your local machine to get up and running quickly. Your computer must be on and connected to the internet when your automated trading program is scheduled to run. I bought a cheap $270 micro computer [KAMRUI GK3 Pro Mini PC](https://www.amazon.com/gp/product/B0BC7S9R5C/ref=ppx_yo_dt_b_asin_title_o00_s01?ie=UTF8&th=1) with Windows 11, specifically for running my trading bot. All it will do is run my trading program (a Python file) at a certain time every day using Windows Task Scheduler. Once I have the time, I'm gonna return to climbing up that steep learning curve for new users of Google Cloud and AWS, platforms I have never used before, this is all totally new for me. I would appreciate any samples and/or examples of you using AWS.
